unitTests/ui/ResultListTest.ts

```diff
@@ -784,11 +784,58 @@ export function ResultListTest() {
         describe('enableScrollToTop set to', () => {
           let infiniteScrollContainer: HTMLElement;
           let allResult: IQueryResults;
+          let scrollToCalled: boolean;
+
           beforeEach(() => {
-            infiniteScrollContainer = { scrollTop: 500 } as HTMLElement;
+            scrollToCalled = false;
+            infiniteScrollContainer = ({
+              scrollTop: 500,
+              scrollTo: (x: number, y: number) => {
+                scrollToCalled = true;
+                infiniteScrollContainer.scrollTop = x;
+              }
+            } as any) as HTMLElement;
             allResult = FakeResults.createFakeResults(50);
           });
 
+          it('should scroll to top on query success', done => {
+            const option: IResultListOptions = {
+              enableInfiniteScroll: true,
+              infiniteScrollContainer: infiniteScrollContainer,
+              enableScrollToTop: true
+            };
+            test = Mock.basicComponentSetup<ResultList>(ResultList, option);
+            test.cmp.currentlyDisplayedResults.push(allResult.results[1]);
+            (test.cmp.queryController.fetchMore as jasmine.Spy).and.returnValue(Promise.resolve(allResult));
+            test.cmp.displayMoreResults(50).then(() => {
+              Simulate.query(test.env);
+              setTimeout(() => {
+                expect(scrollToCalled).toBeTruthy();
+                expect(infiniteScrollContainer.scrollTop).toBe(0);
+                done();
+              }, 0);
+            });
+          });
+
+          it('should scroll to top on query success even in IE11', done => {
+            infiniteScrollContainer.scrollTo = undefined;
+            const option: IResultListOptions = {
+              enableInfiniteScroll: true,
+              infiniteScrollContainer: infiniteScrollContainer,
+              enableScrollToTop: true
+            };
+            test = Mock.basicComponentSetup<ResultList>(ResultList, option);
+            test.cmp.currentlyDisplayedResults.push(allResult.results[1]);
+            (test.cmp.queryController.fetchMore as jasmine.Spy).and.returnValue(Promise.resolve(allResult));
+            test.cmp.displayMoreResults(50).then(() => {
+              Simulate.query(test.env);
+              setTimeout(() => {
+                expect(infiniteScrollContainer.scrollTop).toBe(0);
```

Comment:
Are we sure the `scrollTop` property is not `0` at the beginning? I think it would be good to assert it in the test.

Reply:
We set it in a before each :)